intellij-plugin-structure/tests/src/test/kotlin/com/jetbrains/plugin/structure/domain/ReSharperVersionTest.kt

```diff
@@ -67,6 +66,78 @@ class ReSharperVersionTest {
     Assert.assertEquals("RSU-2021.3", resharperVersion.asString())
   }
 
+  @Test
+  fun getLowerVersionOnlyBaselineTest() {
+    val currentVersion = version("RSU-2023.0.0")
+    val lowerVersion = currentVersion.getLowerVersion()
+    Assert.assertEquals(2022, lowerVersion.components[0])
+    Assert.assertEquals(9, lowerVersion.components[1])
+    Assert.assertEquals(2, lowerVersion.components.size)
+    Assert.assertEquals("RSU-2022.9", lowerVersion.asString())
+  }
+
+  @Test
+  fun getLowerVersionBuildNotZeroTest() {
+    val currentVersion = version("2021.2")
+    val lowerVersion = currentVersion.getLowerVersion()
+    Assert.assertEquals(2021, lowerVersion.components[0])
+    Assert.assertEquals(1, lowerVersion.components[1])
+    Assert.assertEquals(9, lowerVersion.components[2])
+    Assert.assertEquals(3, lowerVersion.components.size)
+    Assert.assertEquals("2021.1.9", lowerVersion.asString())
+  }
+
+  @Test
+  fun getLowerVersionMinorNotZeroTest() {
+    val currentVersion = version("RS-2022.3.1")
+    val lowerVersion = currentVersion.getLowerVersion()
+    Assert.assertEquals(2022, lowerVersion.components[0])
+    Assert.assertEquals(3, lowerVersion.components[1])
+    Assert.assertEquals(0, lowerVersion.components[2])
+    Assert.assertEquals(3, lowerVersion.components.size)
+    Assert.assertEquals("RS-2022.3.0", lowerVersion.asString())
```

Comment:
Is it actually a correct version? Shouldn't it be `2022.2.9`?

Reply:
Also how `lowerVersion` would be used? Would it be correct to use `2022.2.9`? I mean it could be that the latest `2022.2.x` version of Resharper is `2022.2.4`, for example

Reply:
ReSharper range has `isMinIncluded`, `isMaxIncluded` parameters, so when we get, for example, untilLong and max compatible version for update is not included, we need to take a closer lower resharper version. 
I thought about `2022.2.4` or `2022.2.9`, but it seems there is no difference, both of them cover all possible 2022.2 versions, and in case of `2022.2.9` we shouldn't think which minor version is exactly the last.

>Is it actually a correct version? Shouldn't it be `2022.2.9`?
Why do you think it should be `2022.2.9`. The current version is 2022.3.1, so I suppose that the lower should be 2022.3.0. The original compatibility range, for example, is [2022.2, 2022.3.1), so it's changed to [2022.2, 2022.3.0]